### PR TITLE
if def guards, switch to bools, and use c++ struct and enum declaration

### PR DESCRIPTION
**Pull request by jeanlego**
#### Description
add missing ifdef guard
switch boolean macros to stdbool
use c++ style enum and structs

#### Motivation and Context
things are included without being explicitly asked for, and need to remove externed value

#### How Has This Been Tested?
odin pre_commit

#### Types of changes

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: ODIN_II/SRC/include/odin_types.h [hunk 1]
```diff
@@ -186,30 +158,30 @@ extern const char *edge_type_e_STR[];
...
-typedef enum
+enum file_extension_supported
 {
 	VERILOG,
 	file_extension_supported_END
-} file_extension_supported;
+};
 
-typedef enum
+enum bases
```
Comment (ademmings) [inline ODIN_II/SRC/include/odin_types.h]: this is unused now - can be deleted

File: ODIN_II/SRC/include/odin_types.h [hunk 2]
```diff
@@ -186,30 +158,30 @@ extern const char *edge_type_e_STR[];
...
 	HEX = 16,
 	OCT = 8,
 	BIN = 2,
 	LONG = 0,
 	bases_END
-} bases;
+};
 
-typedef enum
+enum signedness
```
Comment (ademmings) [inline ODIN_II/SRC/include/odin_types.h]: can be deleted

File: ODIN_II/SRC/simulate_blif.cpp [hunk 3]
```diff
@@ -3907,17 +3917,12 @@ static void write_vector_to_file(lines_t *l, FILE *file, int cycle)
 
 		if (line_has_unknown_pin(line, cycle) || num_pins == 1)
 		{
-			if ((num_pins + 1) > BUFFER_MAX_SIZE)
```
Comment (ademmings) [inline ODIN_II/SRC/simulate_blif.cpp]: why did you delete these error messages??

Comment (jeanlego) [reply ODIN_II/SRC/simulate_blif.cpp]: No more buffer to overflow, stale check

Comment (jeanlego) [reply ODIN_II/SRC/simulate_blif.cpp]: also if undefined value are x by default
